archiver/helpers.py

```diff
@@ -338,36 +339,36 @@ def file_is_valid_archive_or_terminate(file_path):
         terminate_with_message(f"File {file_path.as_posix()} is not a valid archive of type {COMPRESSED_ARCHIVE_SUFFIX} or {ENCRYPTED_ARCHIVE_SUFFIX} or doesn't exist.")
 
 
-def filename_without_extensions(path):
-    """Removes every suffix, including .partX"""
-    suffixes_string = "".join(path.suffixes)
-
-    return path.name[:-len(suffixes_string)]
+def filename_without_archive_extensions(path):
+    """Removes every archiving suffix"""
+    return filepath_without_archive_extensions(path).name
 
 
-def filepath_without_extensions(path:Path) -> Path:
-    """Removes every suffix, including .partX"""
-    suffixes_string = "".join(path.suffixes)
+def filepath_without_archive_extensions(path:Path) -> Path:
+    """Removes every archiving suffix"""
+    while re.match(ARCHIVE_SUFFIXES_REG, path.suffix):
+        path = path.with_suffix('')
+    return path
 
     return path.parent / path.name[:-len(suffixes_string)]
 
 def infer_source_name(source_path: Path) -> Path:
 
     if not source_path.is_dir():
-        return filepath_without_extensions(source_path)
+        return filepath_without_archive_extensions(source_path)
     else:
         all_files = [p for p in source_path.iterdir() if p.is_file()]
-        unique_names = list(set([filepath_without_extensions(f) for f in all_files]))
+        unique_names = list(set([filepath_without_archive_extensions(f) for f in all_files]))
 
         if len(unique_names) == 0:
             terminate_with_message('There are no archive files present')
         elif len(unique_names) > 1:
-            terminate_with_message(f'More than one possible archive name detected: {str(unique_names)}')
+            terminate_with_message(f'Automatic archive name detection has failed. More than one possible archive name detected: {str(unique_names)}\noptionally use --archive_name to specify archive name.')
 
         return unique_names[0]
 
 
-def filename_without_archive_extensions(path):
+def filename_without_archive_extensions_multipart(path):
```

Comment:
Should we change it to use `ARCHIVE_SUFFIXES_REG`, but without `.parts` and `.part[0-9]+` for consistency?

Reply:
I think it is fine if it removes `.parts.txt`. So we only would need to keep `.part[0-9]+`.

Reply:
I think this would benefit from some generalisation, also with the `COMPRESSED_ARCHIVE_SUFFIX` and the `ENCRYPTED_ARCHIVE_SUFFIX`, but I would like to defer that to a later PR. Feel free to suggest something.